testsuite/classlibrary/TestFunction.sc

```diff
@@ -50,6 +52,8 @@ CommonTestClass {
 	var <>a, <b = 3;
 	classvar <>x, <y = 7;
 
+	*initClass { x = nil; y = 7; }
```

Comment:
why is that necessary?

Reply:
for use in TestFunction.setUp above

Reply:
ah I see. Wouldn't it be better to give this method a different name? The method `*initClass` is called for all classes when the library is recompiled. This method seems to have a slightly different role.

Reply:
yes, there could be a `reset` function called by `TestFunction.setUp` that calls `initClass` , but i think this is splitting hairs w/r/t the design of a mock object. you could argue that the tests are operating under the assumption that they're testing the state of the class after it's been initialized, or that initialization should only be called once, etc. etc.

Reply:
just asking: why not just a class method called `*init` that is called by `*setUp`?

Reply:
OK, if it makes you happy, why not name it init. 
To me, since it _resets_ the _class_ to its _initial_ state, 
reset seems fine, init is fine, and initClass seems fine - having it run on startup is fine, the class should come out of startup in its initial state anyway.

It fixes a test that is unrelated to this  (unfortunately big) PR and only showed up by coincidence. I would be really happy for the PR to be merged ASAP, and going at this level of detail that would take forever ;-)

So in 60997a3, I changed it to init (why not ...), removed the unneeded reset calls, running 
```Testfunction.run``` twice runs fine.

@telephon, if you would like to help with this PR, could you test that clientID/allocation for multiple clients works for you now? 
thanks, adc